### DISCUSSION TITLE
brew --prefix <formula>, but for libexec

### DISCUSSION THREAD

**Discussion by asbjornu** [Homebrew/discussions#568] [Everyday usage]
I want to add a `caveat` to the `dotnet` formula to inform users how to set the `DOTNET_ROOT` variable to the installation path of the formula. Instead of just printing the static path of `#{libexec}` to the terminal, which would require a change to the `export DOTNET_ROOT` statement on every upgrade of the formula, I would prefer the instructions to be something like this:

```shell
export DOTNET_ROOT="$(brew --libexec dotnet)"
```

Now `--libexec` is not a valid argument to the `brew` command, but it would be nice if it was. `brew info` does return the path to `libexec`, but not in isolation so it would require performing `sed` or similar on the output to dig out just the path to `libexec`.

How do I best accomplish this with Homebrew? Is there a best practice for this?

**Comment by SMillerDev** [comment]
Why not `#{opt_libexec}` if it exists? That one won't change with each version.

**Comment by asbjornu** [comment]
That seems to work great, thanks @SMillerDev! I submitted Homebrew/homebrew-core#69323 using `#{opt_libexec}`.
